felix/dataplane/linux/int_dataplane.go

```diff
@@ -1563,6 +1556,42 @@ func (d *InternalDataplane) setUpIptablesBPF() {
 	}
 }
 
+// setUpIptablesBPFEarly that need to be written asap
+func (d *InternalDataplane) setUpIptablesBPFEarly() {
+	rules := bpfMarkPreestablishedFlowsRules()
+
+	for _, t := range d.iptablesFilterTables {
+		// We want to prevent inserting the rules over and over again if something later
+		// crashed. We do not expect that we would inset just a part of the batch as that
+		// should be handled by the iptables-restore transaction.  Never the less if we
+		// see that unexpected case, perhaps due to an upgrade, we skip over updating the
+		// iptables now and will wait for the full resync. That could be temporarily
+		// disrupting.
+		if present := t.CheckRulesPresent("FORWARD", rules); present != nil {
+			if len(present) != len(rules) {
+				log.WithField("present rules", present).
```

Comment:
```suggestion
				log.WithField("presentRules", present).
```
I generally treat these like variable names